### DISCUSSION TITLE
[Tooltip] Usage in iframe with custom container

### DISCUSSION THREAD

**Discussion by syntaxhr** [radix-ui/primitives#1715] [Help]
Hello, I am having a problem using tooltip inside iframes. I just started using your library for building a UI kit and we need to provide support also for usage inside iframes.
I can set a custom container and where it should be attached (in this case the document of the iframe), BUT I am guessing the events that handle opening and closing the tooltip (like mouse enter and leave) are attached to the outer document since the tooltip catches those events in the moment I hover outside the iframe where the tooltip is rendered.
Is there a way I can achieve that or do I need to handle all the events by myself instead?

**Comment by benoitgrelard** [comment]
Hey @syntaxhr,

Could you provide us with a sandbox demonstrating the issue?

Thanks 🙏

**Reply by syntaxhr** [reply]
Hi @benoitgrelard 
here is an example of the problem: https://codesandbox.io/s/exciting-snowflake-c1cnyf?file=/src/App.tsx
You can see how we implement a "providedDocument/Window" and why is the reasoning.
In our app we need to provide usage trough injected applications that are rendered in iframes.
It would be great if all components from radix would accept a "provided document/window" and then attach events on them instead of global instances.
In our current solution we are doing that, but since we would like to cover all accessibility issues trough radix, without that option, we cannot actually use it :(
I mean, we could, but we would have to handle open/close state by ourself by implementing all the events on our side, and basically would beat the point about using radix.

P.S. I think this applies also to other components like dialog, dropdowns, etc...

Thank you for your help..

**Reply by benoitgrelard** [reply]
That seems to be the same issue that was recently patched in `DismissableLayer` (#1677). We know it needs to be addressed globally everywhere we attach document listeners. Essentially they always need to be attached to the `ownerDocument` instead of the global `document`.

I will create a new issue for us to tackle this throughout the entire library. (edit: #1721)

Thanks for raising!

**Reply by syntaxhr** [reply]
Great! and thank you @benoitgrelard 
I know you probably cannot say right away, but do you know approximately when this could be available if you are going to implement it?

**Reply by benoitgrelard** [reply]
I do not know yet no, but I do agree we should look at it with fairly high priority.

**Reply by 2-fly-4-ai** [reply]
> I do not know yet no, but I do agree we should look at it with fairly high priority.

Hey did you guys ever solve this? I am having the same issues trying to render ShadcnUI components built on radix primitives within the iframe container.
